actors/market/src/types.rs

```diff
@@ -100,6 +100,7 @@ pub struct SectorDealData {
 pub struct ActivateDealsParams {
     pub deal_ids: Vec<DealID>,
     pub sector_expiry: ChainEpoch,
+    pub sector_type: RegisteredSealProof,
```

Comment:
Since only one of the call sites needs the CommD result, let's skip it when not needed. Make this `sector_type` optional, set it only from prove_replica_update, and return `None` for commd when the seal proof is not provided.